sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -1184,4 +1186,8 @@ class MapboxMap internal constructor(
     }
     return null
   }
+
+  internal fun clearListeners() {
```

Comment:
Do we even need that function?
`awaitingStyleGetters` are also internal, we could clear them directly.

Reply:
This is to make sure we don't get into the following:
```
 - user startups a map and loads a style
 - exists activity before style has loaded
 - listener is leaked
```